### ISSUE TITLE
[10.0] POS Orders - POS Stock Picking Stuck in Available State: Validate Picking >> Record does not exist or has been deleted

### ISSUE THREAD

**Issue by lukebranch** [odoo/odoo#21932]
<h1>Impacted versions:</h1>
10.0

<h1>Steps to reproduce:</h1>
Unable to provide at this point in time (will update this ticket with more information when I am able to provide steps to reproduce). <br/>

<h1>Current behaviour:</h1>
Periodically the Odoo service will crash while using the POS (this may or may not be related), and the link between stock move, and stock pack operation is lost, leaving Inventory > Dashboard > POS Orders > (Stock Picking) stuck in the 'Available' state (or some state other than 'Done').<br/>

<b>Chatter Log Always Shows:</b> Status: Partially Available  Available

<h1>Expected behaviour:</h1>
Odoo will process the Delivery Order completely for all POS orders.

<h1>Current Workaround:</h1>

1. Install stock_cancel module (see zip below for reference - based on https://github.com/OCA/stock-logistics-workflow/tree/10.0/stock_cancel)
2. Cancel this picking
3. Run SQL to remove current data (where picking_id = 1234 is an example picking):
`delete from stock_pack_operation where picking_id = 1234;
delete from stock_move_operation_link where  move_id in (select id from stock_move where picking_id = 1234);`
4. Reopen this picking and validate it

<h1>Video/Screenshot link (optional):</h1>
<img src="https://user-images.githubusercontent.com/2337666/34451498-a42b3fd0-ed62-11e7-936d-a717c4f5c45f.gif"/>

<h1>Customized Stock Cancel Module:</h1>
<a href="https://github.com/odoo/odoo/files/1594176/stock_cancel.zip">https://github.com/odoo/odoo/files/1594176/stock_cancel.zip</a>

**Comment by lukebranch** [comment]
We've had this issue crop up again with a group of 7 orders all within the space of one day.

Here's an explanation from our developer:

1. stock.pack.operation
2. stock.move
3. stock.move.operation.link

object 1 and 2 link each other via object 3
on object 3, we have 2 fields: move_id and operation_id.
Example: in this case, object 3 have 1 record: move_id=110 and operation_id=100
in object 2, we can find record with ID = 110, but in object 1, ID 100 is not exist

Basically what is happening is Odoo seems to be looking for records that don't exist. We can't find steps to reproduce however I suspect one of the following may be the issue:

pos_cache is installed and is set to refresh once every 15 minutes (they have roughly 1000 products)

I am seeing an error regularly in /var/log/postgresql/postgresql.log

could not obtain lock on row in relation "ir_cron"

Has anyone else encountered this issue?

**Comment by Yenthe666** [comment]
Dear @lukebranch,

As Odoo 13 has been released last week it means that our support for Odoo version 10 has been stopped.
If you can still reproduce this issue on a newer version (version 11, 12 or 13 at the point of writing) then please do respond back or create a new issue.
Sorry that we didn't get around to resolving your issue.

Regards,
Yenthe
